Updating from 1.0.0 to 1.0.3 causes TS Error TS2345: Argument of type

I believe I am forming my theme incorrectly, however with version 1.0.0 this error does not happen.

```
export const colors = {
  grey: {
    100: "#e3e3e3",
    200: "#edf2f7",
    300: "#e2e8f0",
    400: "#cbd5e0",
    light: "#D4D4D4",
    medium: "#929292",
    dark: "#1D1D1B",
    navSubMenu: "#9F9F9F",
  },
  white: "#fff",
  black: {
    25: "rgba(0, 0, 0, 0.25)",
    50: "rgba(0, 0, 0, 0.5)",
    100: "#000",
    150: "#1D1D1B",
  },
  pink: {
    dark: "#FF8FA2",
    base: "#f6ced6",
    light: "#FAA4B3",
    lighter: "#F6CED6",
  },
}

```

theme/index.ts:34:34 - error TS2345: Argument of type '{ fonts: { demi: string; book: string; light: string; }; fontWeights: { normal: number; medium: number; bold: number; light: number; }; fontSizes: { 13: string; 8: string; 5: string; 3: string; 2: string; 1: string; 0: string; "-1": string; "-2": string; }; ... 6 more ...; components: { ...; }; }' is not assignable to parameter of type 'DeepThemeExtension<{ components: { Accordion: { parts: string[]; baseStyle: { container: { borderTopWidth: string; borderColor: string; _last: { borderBottomWidth: string; }; }; button: { fontSize: string; _focus: { boxShadow: string; }; _hover: { ...; }; _disabled: { ...; }; px: number; py: number; }; panel: { ...;...'.
  Type '{ fonts: { demi: string; book: string; light: string; }; fontWeights: { normal: number; medium: number; bold: number; light: number; }; fontSizes: { 13: string; 8: string; 5: string; 3: string; 2: string; 1: string; 0: string; "-1": string; "-2": string; }; ... 6 more ...; components: { ...; }; }' is not assignable to type 'Partial<ThemeExtensionTypeHints>'.
    Types of property 'colors' are incompatible.
      Type '{ grey: { 100: string; 200: string; 300: string; 400: string; light: string; medium: string; dark: string; navSubMenu: string; }; white: string; black: { 25: string; 50: string; 100: string; 150: string; }; pink: { ...; }; }' is not assignable to type 'Record<string, string | Partial<ColorHues>>'.
        Property 'pink' is incompatible with index signature.
          Type '{ dark: string; base: string; light: string; lighter: string; }' is not assignable to type 'string | Partial<ColorHues>'.
            Type '{ dark: string; base: string; light: string; lighter: string; }' has no properties in common with type 'Partial<ColorHues>'.

34 export const theme = extendTheme(appTheme)
                                    ~~~~~~~~


Found 1 error.

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
husky > pre-commit hook failed (add --no-verify to bypass)

--- a/.changeset/young-trainers-invite.md
+++ b/.changeset/young-trainers-invite.md
@@ -1,0 +1,5 @@
+---
+"@chakra-ui/react": patch
+---
+
+fix(extend-theme): allow string properties besides ColorHue in colors

--- a/packages/react/src/extend-theme.ts
+++ b/packages/react/src/extend-theme.ts
@@ -3,7 +3,7 @@ import { isFunction, mergeWith } from "@chakra-ui/utils"
 import { ColorHues } from "@chakra-ui/theme/dist/types/foundations/colors"
 
 type ThemeExtensionTypeHints = {
-  colors: Record<string, Partial<ColorHues> | string> // typehints for color definitions
+  colors: Record<string, Partial<ColorHues> | Record<string, string> | string> // typehints for color definitions
 }
 
 /**

--- a/packages/react/tests/extend-theme.test.tsx
+++ b/packages/react/tests/extend-theme.test.tsx
@@ -66,4 +66,112 @@ describe("extendTheme", () => {
     // should have more properties from the default theme
     expect(Object.keys(solidStyles).length).toBeGreaterThan(1)
   })
+
+  it("should pass typescript lint with random custom theme", () => {
+    const override = {
+      shadows: {
+        outline: "0 0 0 3px rgb(0, 255, 0)",
+      },
+      colors: {
+        gray: {
+          "300": "red",
+        },
+        myCustomColor: {
+          "50": "papayawhip",
+        },
+      },
+      textStyles: {
+        dl: {
+          lineHeight: "tall",
+          "dd + dt": {
+            marginTop: "4",
+          },
+        },
+        dt: {
+          color: "gray.600",
+          fontSize: "sm",
+          fontWeight: "bold",
+        },
+        dd: {
+          color: "gray.800",
+        },
+      },
+      config: {
+        useSystemColorMode: false,
+        initialColorMode: "dark",
+      },
+      styles: {
+        global: {
+          body: {
+            color: "green.500",
+          },
+        },
+      },
+      components: {
+        Button: {
+          variants: {
+            ghost: {
+              fontFamily: "mono",
+              _active: {
+                bg: "red",
+              },
+            },
+          },
+        },
+        Input: {
+          baseStyle: {
+            field: {
+              outline: "1px solid black",
+            },
+          },
+          variants: {
+            outline: () => ({
+              field: {
+                bg: "green.500",
+              },
+            }),
+            myCustomVariant: {
+              field: {
+                bg: "red.500",
+              },
+            },
+          },
+        },
+      },
+    }
+
+    extendTheme(override)
+  })
+
+  it("should pass typescript lint with non colorhue properties", () => {
+    const override = {
+      colors: {
+        grey: {
+          100: "#e3e3e3",
+          200: "#edf2f7",
+          300: "#e2e8f0",
+          400: "#cbd5e0",
+          light: "#D4D4D4",
+          medium: "#929292",
+          dark: "#1D1D1B",
+          navSubMenu: "#9F9F9F",
+        },
+        white: "#fff",
+        black: {
+          25: "rgba(0, 0, 0, 0.25)",
+          50: "rgba(0, 0, 0, 0.5)",
+          100: "#000",
+          150: "#1D1D1B",
+        },
+        pink: {
+          dark: "#FF8FA2",
+          base: "#f6ced6",
+          light: "#FAA4B3",
+          lighter: "#F6CED6",
+        },
+      },
+    }
+
+    extendTheme(override)
+  })
 })

